### PR TITLE
docs: update catalog error eventing/logging

### PR DESCRIPTION
**Pull request by codingdiaz**
## Hey, I just made a Pull Request!

This updates the documentation to reflect the changes around catalog error logging. https://github.com/backstage/backstage/pull/23022

I wasn’t sure if adding this guide here made sense or not but I wasn’t quite sure the best place to put this content. i.e. should this section be updated and linked out to a guide. 

I also feel like there is a much cleaner way (with generics) to define the shape of the catalog event. I was getting type errors when trying something like this:

```
    params: EventParams<{
      entity: string;
      location?: string;
      errors: Array<Error>;
    }>,
```

I also didn’t know if a simple module would help for users that want to turn these logs on rather than creating their own? I’m happy to contribute that!

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/features/software-catalog/life-of-an-entity.md
Comment (awanlin) [inline docs/features/software-catalog/life-of-an-entity.md:291]: I think we should drop this off, Adopters should be on the new backend system, it's been the default for some time now

File: docs/features/software-catalog/life-of-an-entity.md
Comment (awanlin) [inline docs/features/software-catalog/life-of-an-entity.md:282]: Thought: Would it make sense as a module so you don't need to add all the code just the module?

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @codingdiaz, really appreciate it 🚀 

Left a few comments, nothing major, I do think that maybe this should be somewhere else though. This page is great as an overview and I like that it's very focused on the specific topic only. Could we move this here: https://backstage.io/docs/features/software-catalog/configuration

File: docs/features/software-catalog/life-of-an-entity.md
Comment (codingdiaz) [reply docs/features/software-catalog/life-of-an-entity.md:282]: I actually put that in the PR description as a thought! I’m happy to do that in this PR. I was maybe slightly confused on what to name it. i.e is this a catalog module or an events module?

File: docs/features/software-catalog/life-of-an-entity.md
Comment (codingdiaz) [reply docs/features/software-catalog/life-of-an-entity.md:223]: Nice catch/feedback!

Comment (codingdiaz) [reply docs/features/software-catalog/life-of-an-entity.md]: @awanlin I got this updated.

File: docs/features/software-catalog/life-of-an-entity.md
Comment (codingdiaz) [reply docs/features/software-catalog/life-of-an-entity.md]: @awanlin that sounds good, I dropped the docs.

**Comment by codingdiaz** [comment]
> Thanks for this @codingdiaz, really appreciate it 🚀
> 
> Left a few comments, nothing major, I do think that maybe this should be somewhere else though. This page is great as an overview and I like that it's very focused on the specific topic only. Could we move this here: https://backstage.io/docs/features/software-catalog/configuration

Any time! I just updated the docs, I agree with you and I linked to the configuration section.

File: docs/features/software-catalog/life-of-an-entity.md
Comment (codingdiaz) [reply docs/features/software-catalog/life-of-an-entity.md]: I just added a module since it’s a really low lift and I think that makes a lot of sense.

File: .changeset/ten-pots-walk.md
Comment (awanlin) [inline .changeset/ten-pots-walk.md:7]: Can we replace this with a link to the docs you added on this topic?

**Review by awanlin — CHANGES_REQUESTED** [review]
Just gave the docs parts a look. Left some comments, should be pretty easy to address. 

Did not look at the code. I was mostly suggesting it should be a module, was hoping the maintainers would give us a quick heads up on if that's the right direction.

File: docs/features/software-catalog/configuration.md
Comment (codingdiaz) [reply docs/features/software-catalog/configuration.md:225]: Ah, yeah I can be more specific. For example, you might want to send a Slack message to the person who’s file has an error and isn’t registering updates in Backstage. Or, you might want to make a JIRA ticket for a team to look into the warnings/errors.

File: plugins/catalog-backend-module-logs/package.json
Comment (freben) [inline plugins/catalog-backend-module-logs/package.json:6]:
```suggestion
    "role": "backend-plugin-module",
    "pluginId": "catalog",
    "pluginPackage": "@backstage/plugin-catalog-backend"
```

since the 1.28 release, we now expect an extra set of metadata in all plugin packages - this is the effect that running `yarn fix` would get, so now hopefully you don't have to rebase and do it yourself 😅

File: plugins/catalog-backend-module-logs/package.json
Comment (freben) [inline plugins/catalog-backend-module-logs/package.json:34]:
```suggestion
```

this dependency is not being used

**Review by freben — APPROVED** [review]
Nice!

File: docs/features/software-catalog/life-of-an-entity.md
Comment (codingdiaz) [reply docs/features/software-catalog/life-of-an-entity.md:207]: I removed that second sentence, I definitely missed that. As a backstage adopter/maintainer I was thrown off that these logs were turned off (hard to see it was done) so I think the context will be helpful.

**Comment by codingdiaz** [comment]
@freben / @ahhhndre. I’m getting some errors checking the API reports but locally I’m running `yarn build:api-reports` & then `yarn build:api-reports:only --ci —docs` & `echo $` with an output of 0. 

Am I missing something to get the api report checks passing in CI?

**Review by awanlin — APPROVED** [review]
Thanks again for this @codingdiaz 🚀

**Comment by freben** [comment]
Funky - i think it needed a bit of a rebase and massage. Managed to generate the report properly now and pushed. Let's see how it goes.

**Comment by freben** [comment]
UNLESS I PRESS THE WRONG BUTTON ACCIDENTALLY FIRST

**Comment by codingdiaz** [comment]
@freben I’m all set with changes on this one (I don’t have the ability to merge).

**Comment by awanlin** [comment]
@vinzscam looks like we need your approval here too ;)

**Pull request merged**
